### PR TITLE
feat(payment): INT-4592 Added 3DS handler to SquareV2

### PR DESCRIPTION
**Pull request by JoseAlvarezHdz**
## What? [INT-4592](https://jira.bigcommerce.com/browse/INT-4592)

Added VerifyBuyer from SquareV2 Script Client to SquareV2 Payment Strategy

## Why?

In Order for SquareV2 to Support 3DS 

## Testing / Proof

Screen Shot 2021-07-20 at 12 17 17

## Depends on 

https://github.com/bigcommerce/bigcommerce/pull/42035

@bigcommerce/checkout @bigcommerce/payments @bigcommerce/apex-integrations

### CODE REVIEW

**Review by FranciscoOnt — APPROVED** [review]
LGTM

File: src/payment/strategies/square/square-form.ts [hunk 1]
```diff
@@ -2,10 +2,36 @@ export default interface SquarePaymentForm {
...
+export interface SquareVerificationResult {
+    token: string;
+    userChallenged: boolean;
+}
+
+export enum SquareIntent {
+    CHARGE = 'CHARGE',
+    STORE = 'STORE',
+}
+export interface VerificationDetails {
```
[the remark this answers is not in the document]
Comment (JoseAlvarezHdz) [reply src/payment/strategies/square/square-form.ts:25]: Done! thanks!

File: src/payment/strategies/square/square-payment-strategy.ts [hunk 2]
```diff
@@ -158,16 +167,23 @@ export default class SquarePaymentStrategy implements PaymentStrategy {
...
+        if (!this._squareOptions) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._squareOptions;
+    }
+
     private _handleWalletNonceResponse(
-        errors?: NonceGenerationError[],
+        errors?: NonceGenerationError[] | undefined,
```
Comment (bc-akshatkaushik) [inline src/payment/strategies/square/square-payment-strategy.ts:179]: If the errors field is optional why do we need to add a union type of undefined?

File: src/payment/strategies/square/square-payment-strategy.ts [hunk 3]
```diff
@@ -281,9 +286,65 @@ export default class SquarePaymentStrategy implements PaymentStrategy {
...
+            phone: billingAddress.phone,
+        };
+    }
+
+    private _getAmountAndCurrencyCode(): string[] {
+        const state = this._store.getState();
+        const storeConfig = state.config.getStoreConfigOrThrow();
+        const checkout = state.checkout.getCheckoutOrThrow();
+
+        return [String(checkout.grandTotal), storeConfig.currency.code];
```
Comment (bc-akshatkaushik) [inline src/payment/strategies/square/square-payment-strategy.ts:313]: Shouldn't the currency code be fetched from Cart config instead of store config?

**Comment by JoseAlvarezHdz** [comment]
@bc-akshatkaushik i made the changes you suggest, do you mind reviewing it again please!

**Review by bc-mackxu — APPROVED** [review]
LGTM, Thank you.

**Pull request merged**